### PR TITLE
WIP Make default executable Platform independen

### PR DESCRIPTION
**Pull request by lukaslueg**
Here is some code to make the default executable discoverable on multiple platforms. It allows to execute all tests on Linux, which pass (besides the integration tests, which have to go anyway).

`LaunchOptions` inevitably loses the ability to implement `Default`, because we may not be able to (somewhat) safely discover a default path. It is up to the user (and highly platform dependent) what to do then.

### CODE REVIEW

File: src/process.rs [hunk 1]
```diff
@@ -130,7 +171,7 @@ impl Process {
         }
 
         let process = TemporaryProcess(
-            Command::new(launch_options.path)
+            Command::new(&launch_options.path)
```
Comment (atroche) [inline src/process.rs]: Will this panic on systems where the binary doesn't exist? I'm not sure we ever want to let users construct a `LaunchOptions` where `path` is `None`.

File: src/process.rs [hunk 2]
```diff
@@ -39,21 +39,62 @@ impl Drop for TemporaryProcess {
...
+
+    pub fn headless(mut self, headless: bool) -> Self {
+        self.headless = headless;
+        self
+    }
+
+    pub fn port(mut self, port: Option<u16>) -> Self {
+        self.port = port;
+        self
     }
```
Comment (atroche) [inline src/process.rs]:
```suggestion
    }
    
    pub fn path(mut self, path: std::path::PathBuf) -> Self {
        self.path = path;
        self
    }
```

For when users want everything default except for the path (e.g. sometimes I switch between versions of Chrome).

File: src/process.rs (hunk 1 above)
Comment (lukaslueg) [reply src/process.rs]: `path` can't be `None` here. If you have a `LaunchOption`, you have _some_ `path`. That's why `LaunchOption` no longer implements `Default`: In order to be able to return `None` if there is no default.

Comment (atroche) [reply src/process.rs]: oh, I see, thanks

File: src/process.rs [hunk 3]
```diff
@@ -39,21 +39,67 @@ impl Drop for TemporaryProcess {
...
+    pub fn default_executable() -> Option<std::path::PathBuf> {
+        // TODO BSDs/Unixes are the same?
+        #[cfg(target_os = "linux")]
+        {
+            // TODO Look at $BROWSER and if it points to a chrome binary
+            // $BROWSER may also provide default arguments, which we may
+            // or may not override later on.
+
+            // TODO More paths!?
+            let default_paths = &["/usr/bin/google-chrome-stable", "/usr/bin/chromium-browser"][..];
```
Comment (atroche) [inline src/process.rs]: Not suggesting we do it in this PR, but we could also search PATH.

**Comment by atroche** [comment]
@lukaslueg I'm happy to merge this, just wanted to check with you first because of the WIP in the title

**Comment by atroche** [comment]
(added you as a collaborator, so merge when ready)

File: src/process.rs [hunk 4]
```diff
@@ -39,21 +39,67 @@ impl Drop for TemporaryProcess {
...
+                    return Some(path.into());
+                }
+            }
+        }
+        // TODO Windows
+        // Check HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\App Paths\chrome.exe
+        None
+    }
+
+    pub fn default() -> Option<Self> {
```
Comment (atroche) [inline src/process.rs]: How would you feel about this being a `Result` instead? If someone calls `LaunchOptions::default()` and gets back `None`, they might not understand why, or what they need to do to fix it.

Comment (atroche) [reply src/process.rs]: And it's also let construction be like this:

`Browser::new(LaunchOptions::default()?)?`

Because you can't use the `?` for `Result`s and `Option`s in the same function, right?

Comment (lukaslueg) [reply src/process.rs]: I've added a default constructor to `Browser`, so the `Browser::new(LaunchOptions::default()?)?` can be written as `Browser::default()?`

**Comment by lukaslueg** [comment]
I merge to master now, can't

**Comment by atroche** [comment]
Not quite sure what you mean. Can you rephrase?

But I'll merge anyway :)
